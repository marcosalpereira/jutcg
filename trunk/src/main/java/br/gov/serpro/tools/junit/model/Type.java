package br.gov.serpro.tools.junit.model;

public class Type {
	private String name;

	private String variableName;

	private boolean primitive;

	private String fullName;

	public String getNewValue() {
		if (!isPrimitive()) {
			if (isCollection()) {
				return String.format("new %s()", getCollectionImpl());
			}
			return String.format("new %s(1)", getName());
		}

		if (getName().equals("boolean")) {
			return "false";
		}

		return "0";
	}

	private String getCollectionImpl() {
		if (isSet()) return "Hash" + name;
		if (isList()) return "Array" + name;
		return name;
	}

	private boolean isCollection() {
		return isList() || isSet();
	}

	private boolean isSet() {
		return name.startsWith("Set");
	}

	private boolean isList() {
		return name.startsWith("List");
		}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
		this.variableName = defineVariableName();
	}

	public void setPrimitive(boolean primitive) {
		this.primitive = primitive;
	}

	public boolean isPrimitive() {
		return primitive;
	}

	public String getVariableName() {
		return variableName;
	}

	private String defineVariableName() {

		if (name == null) {
			return null;
		} else {
			return name.substring(0, 1).toLowerCase() + name.substring(1);
		}
	}

	/** {@inheritDoc} */
	@Override
	public String toString() {
		return name;
	}

	public void setFullName(String fullName) {
		this.fullName = fullName;
	}

	public String getFullName() {
		return fullName;
	}



}
