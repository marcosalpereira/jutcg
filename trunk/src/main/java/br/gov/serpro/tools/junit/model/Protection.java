package br.gov.serpro.tools.junit.model;

/**
 * A java member visibility restriction.
 */
public enum Protection {
    DEFAULT, PRIVATE, PROTECTED, PUBLIC;

    @Override
    public String toString() {
        return this.name().toLowerCase();
    }
}
