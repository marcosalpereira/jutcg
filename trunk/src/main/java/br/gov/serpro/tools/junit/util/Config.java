package br.gov.serpro.tools.junit.util;

import java.util.MissingResourceException;
import java.util.ResourceBundle;

/**
 * Config.
 */
public final class Config {

    /**
     * Resource bundle.
     */
    private static final ResourceBundle RESOURCE_BUNDLE = ResourceBundle
            .getBundle("config");

    /**
     * Construtor.
     */
    private Config() {
    }

    /**
     * Get an property.
     * @param key key
     * @return the value of property
     */
    public static String getString(final String key) {
        try {
            return RESOURCE_BUNDLE.getString(key);
        } catch (final MissingResourceException e) {
            return null;
        }
    }
}
