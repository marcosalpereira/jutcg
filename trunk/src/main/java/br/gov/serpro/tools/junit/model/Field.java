package br.gov.serpro.tools.junit.model;

import java.util.ArrayList;
import java.util.List;

import br.gov.serpro.tools.junit.GeneratorHelper;


public class Field {
	/**
	 * Field name.
	 */
	private String name;

	/**
	 * Field type.
	 */
	private Type type;

	/**
	 * Is field static
	 */
	private boolean ztatic;

	/**
	 * Field's annotations.
	 */
	private List<String> annotations = new ArrayList<String>();

	/**
	 * Field protection.
	 */
	private Protection protection;

	public void setProtection(Protection protection) {
		this.protection = protection;
	}

	public Protection getProtection() {
		return protection;
	}

	public List<String> getAnnotations() {
		return annotations;
	}

	public void setAnnotations(List<String> annotations) {
		this.annotations = annotations;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public Type getType() {
		return type;
	}

	public void setType(Type type) {
		this.type = type;
	}

	public String variableNameForType() {
		if (getType() == null) return null;
		return getType().getVariableName();
	}

	public boolean isStatic() {
		return ztatic;
	}

	public void setStatic(boolean ztatic) {
		this.ztatic = ztatic;
	}

	public boolean isPrivate() {
		return protection == Protection.PRIVATE;
	}
	
	@Override
	public String toString() {
		return this.name;
	}

	public String getSetter() {
		return "set" + GeneratorHelper.upperCaseFirstChar(name);
	}

	public String getGetter() {
		if ("boolean".equals(this.getType().getName())) {
			return "is" + GeneratorHelper.upperCaseFirstChar(name);
		}
		return "get" + GeneratorHelper.upperCaseFirstChar(name);
	}
}
