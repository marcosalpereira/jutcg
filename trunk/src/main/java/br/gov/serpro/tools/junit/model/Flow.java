package br.gov.serpro.tools.junit.model;

import java.util.ArrayList;
import java.util.List;
import java.util.SortedSet;
import java.util.TreeSet;

public class Flow {

	/**
	 * Flow name.
	 */
	private String name = "";

	/**
	 * Invocations for this flow.
	 */
	private List<FieldMethodInvocation> invocations = new ArrayList<FieldMethodInvocation>();

	/**
	 * Method that this flow belongs to.
	 */
	private Method method;

	/**
	 * Fields read on this flow.
	 */
	private SortedSet<Field> readFields = new TreeSet<Field>();

	/**
	 * Fields written on this flow.
	 */
	private SortedSet<Field> writtenFields = new TreeSet<Field>();


	public SortedSet<Field> getWrittenFields() {
		return writtenFields;
	}

	public void setWrittenFields(SortedSet<Field> writtenFields) {
		this.writtenFields = writtenFields;
	}

	public SortedSet<Field> getReadFields() {
		return readFields;
	}

	public void setReadFields(SortedSet<Field> usedFields) {
		this.readFields = usedFields;
	}

	public List<FieldMethodInvocation> getInvocations() {
		return invocations;
	}

	public void setInvocations(List<FieldMethodInvocation> invocations) {
		this.invocations = invocations;
	}

	public Method getMethod() {
		return this.method;
	}

	public void setMethod(Method method) {
		this.method = method;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	/**
	 * Retorna o metodo que foi usado num return neste fluxo.
	 * @return o metodo ou <code>null</code> se nao existir
	 */
	public Method getReturnInvocationMethod() {
		for (FieldMethodInvocation invocation : invocations) {
			if (invocation.isReturnInvocation() ) return invocation.getMethod();
		}
		return null;
	}



}
