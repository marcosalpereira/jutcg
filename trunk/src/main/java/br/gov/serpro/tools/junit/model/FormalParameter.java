package br.gov.serpro.tools.junit.model;

import java.util.List;

/**
 * Formal parameters of a method.
 */
public class FormalParameter {
    /**
     * Formal parameter name.
     */
    private String variableId;

    /**
     * Field type.
     */
    private Type type;

    /**
     * Field's annotations.
     */
    private List<Type> annotations;

    public List<Type> getAnnotations() {
        return this.annotations;
    }

    public void setAnnotations(final List<Type> annotations) {
        this.annotations = annotations;
    }

    public String getVariableId() {
        return this.variableId;
    }

    public void setVariableId(final String name) {
        this.variableId = name;
    }

    public Type getType() {
        return this.type;
    }

    public void setType(final Type type) {
        this.type = type;
    }

}
