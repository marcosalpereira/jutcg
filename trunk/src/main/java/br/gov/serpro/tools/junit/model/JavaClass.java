package br.gov.serpro.tools.junit.model;

import java.util.ArrayList;
import java.util.List;

import br.gov.serpro.tools.junit.Config;

/**
 * A class representing a Java source file.
 */
public class JavaClass {
	/**
	 * Package name.
	 */
	private String packageName;

	/**
	 * Class type.
	 */
	private Type type;

	/**
	 * Class imports.
	 */
	private List<String> imports = new ArrayList<String>();

	/**
	 * Class methods.
	 */
	private List<Method> methods = new ArrayList<Method>();

	/**
	 * Class fields.
	 */
	private List<Field> fields = new ArrayList<Field>();

	public String getPackageName() {
		return packageName;
	}

	public void setPackageName(String packageName) {
		this.packageName = packageName;
	}

	public Type getType() {
		return type;
	}

	public void setType(Type type) {
		this.type = type;
	}

	public List<String> getImports() {
		return imports;
	}

	public void setImports(List<String> imports) {
		this.imports = imports;
	}

	public List<Method> getMethods() {
		return methods;
	}

	public void setMethods(List<Method> methods) {
		this.methods = methods;
	}

	public List<Field> getFields() {
		return fields;
	}

	public void setFields(List<Field> fields) {
		this.fields = fields;
	}

	public boolean isAtView() {
		return getPackageName().startsWith(Config.getString("packagePrefix.view"));
	}

	public boolean isAtDao() {
		return getPackageName()
				.startsWith(Config.getString("packagePrefix.dao"));
	}

	public String variableNameForType() {
		if (getType() == null) return null;
		return getType().getVariableName();
	}

	public Field searchField(String fieldName) {
		for (Field f : getFields()) {
			if (f.getName().equals(fieldName)) return f;
		}
		return null;
	}

	public Method searchMethod(String methodName) {
		for (Method m : getMethods()) {
			if (m.getName().equals(methodName)) return m;
		}
		return null;
	}

}
