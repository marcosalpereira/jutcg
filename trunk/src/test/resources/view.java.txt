package br.gov.esaf.sgc.view.correcao;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

import javax.faces.component.UIComponent;
import javax.faces.component.UIInput;
import javax.faces.context.FacesContext;
import javax.faces.validator.ValidatorException;

import org.jboss.seam.ScopeType;
import org.jboss.seam.annotations.In;
import org.jboss.seam.annotations.Name;
import org.jboss.seam.annotations.Scope;

import br.gov.esaf.sgc.facade.DigitacaoNotaManualFacade;
import br.gov.esaf.sgc.model.CampoAtuacao;
import br.gov.esaf.sgc.model.Cargo;
import br.gov.esaf.sgc.model.Cidade;
import br.gov.esaf.sgc.model.Concurso;
import br.gov.esaf.sgc.model.Inscricao;
import br.gov.esaf.sgc.model.OrgaoContratante;
import br.gov.esaf.sgc.model.TipoConsultaNome;
import br.gov.esaf.sgc.model.Uf;
import br.gov.esaf.sgc.model.prova.Prova;
import br.gov.esaf.sgc.model.prova.ConteudoProva.TipoConteudoProva;
import br.gov.esaf.sgc.model.resultado.ResultadoProvaManual.VersaoResultadoProvaManual;
import br.gov.serpro.util.StringHelper;
import br.gov.serpro.view.util.MessageHelper;

/**
 * Bean para o caso de uso Digitar Resultado Manual.
 */
@Name("digitacaoNotaManualListarBean")
@Scope(ScopeType.CONVERSATION)
public class DigitacaoNotaManualListarBean {

    /**
     * Id do componente com o numero da inscricao na tela.
     */
    static final String ID_INPUT_NUMERO_INSCRICAO = "numeroInscricao";

    /**
     * Id do componente com o nome do candidato na tela.
     */
    static final String ID_INPUT_NOME_CANDIDATO = "nomeCandidato";

    /**
     * Concurso selecionado.
     */
    private Concurso concursoSelecionado;

    /**
     * Prova selecionada.
     */
    private Prova provaSelecionada;

    /**
     * Versão da nota selecionada.
     */
    private VersaoResultadoProvaManual versaoSelecionada;

    /**
     * Numero de Inscricao.
     */
    private String numeroInscricao;

    /**
     * Nome Candidato.
     */
    private String nomeCandidato;

    /**
     * Lista de concursos.
     */
    private List<Concurso> concursos;

    /**
     * Lista de provas.
     */
    private List<Prova> provas;

    /**
     * Lista de versões de nota.
     */
    private List<VersaoResultadoProvaManual> versoes;

    /**
     * Tipo de consulta pelo nome.
     */
    private TipoConsultaNome tipoConsultaNome = TipoConsultaNome.INICIO;

    /**
     * Flag que indica se deve ser mostrada a selecao de localidade de provas.
     */
    private boolean mostrarLocalidades;

    /**
     * Uf Selecionada.
     */
    private Uf ufSelecionada;


    /**
     * Lista de UFs.
     */
    private List<Uf> ufs;

    /**
     * Cidade Selecionada.
     */
    private Cidade cidadeSelecionada;

    /**
     * Cidades.
     */
    private List<Cidade> cidades;

    /**
     * Inscrições Pesquisadas.
     */
    private List<Inscricao> inscricoes;

    /**
     * Map indicando se as inscricoes sao excluiveis.
     */
    private Map<Inscricao, Boolean> inscricoesExcluiveis;

    /**
     * Cargos vinculados ao orgao/concurso/prova.
     */
    private List<Cargo> cargos;

    /**
     * Cargo selecionado.
     */
    private Cargo cargoSelecionado;

    /**
     * Cargos vinculados ao orgao/concurso/prova.
     */
    private List<CampoAtuacao> camposAtuacao;

    /**
     * Cargo selecionado.
     */
    private CampoAtuacao campoAtuacaoSelecionado;

    /**
     * Orgaos vinculados ao concurso/prova.
     */
    private List<OrgaoContratante> orgaos;

    /**
     * Orgao selecionado.
     */
    private OrgaoContratante orgaoSelecionado;

    /**
     * Fachada Digitacao Resultado Nota Manual.
     */
    @In("digitacaoNotaManualFacade")
    private transient DigitacaoNotaManualFacade digitacaoNotaManualFacade;
    
    /**
     * Usuário logado.
     */
    @In
    private UsuarioLogado usuarioLogado;    

    public void setDigitacaoNotaManualFacade(
        DigitacaoNotaManualFacade digitacaoNotaManualFacade) {
        this.digitacaoNotaManualFacade = digitacaoNotaManualFacade;
    }

    public void setCargos(List<Cargo> cargos) {
        this.cargos = cargos;
    }

    public Cargo getCargoSelecionado() {
        return cargoSelecionado;
    }

    public void setCargoSelecionado(Cargo cargoSelecionado) {
        this.cargoSelecionado = cargoSelecionado;
    }

    public void setCamposAtuacao(List<CampoAtuacao> camposAtuacao) {
        this.camposAtuacao = camposAtuacao;
    }

    public CampoAtuacao getCampoAtuacaoSelecionado() {
        return campoAtuacaoSelecionado;
    }

    public void setCampoAtuacaoSelecionado(CampoAtuacao campoAtuacaoSelecionado) {
        this.campoAtuacaoSelecionado = campoAtuacaoSelecionado;
    }

    public void setOrgaoSelecionado(OrgaoContratante orgaoSelecionado) {
        this.orgaoSelecionado = orgaoSelecionado;
    }

    public OrgaoContratante getOrgaoSelecionado() {
        return orgaoSelecionado;
    }

    public void setOrgaos(List<OrgaoContratante> orgaos) {
        this.orgaos = orgaos;
    }

    public TipoConsultaNome getTipoConsultaNome() {
        return tipoConsultaNome;
    }

    public void setTipoConsultaNome(TipoConsultaNome tipoConsultaNome) {
        this.tipoConsultaNome = tipoConsultaNome;
    }

    public Concurso getConcursoSelecionado() {
        return concursoSelecionado;
    }

    public void setConcursoSelecionado(Concurso concursoSelecionado) {
        this.concursoSelecionado = concursoSelecionado;
    }

    public Prova getProvaSelecionada() {
        return provaSelecionada;
    }

    public void setProvaSelecionada(Prova provaSelecionada) {
        this.provaSelecionada = provaSelecionada;
    }

    public VersaoResultadoProvaManual getVersaoSelecionada() {
        return versaoSelecionada;
    }

    public void setVersaoSelecionada(VersaoResultadoProvaManual versaoSelecionada) {
        this.versaoSelecionada = versaoSelecionada;
    }

    public String getNumeroInscricao() {
        return numeroInscricao;
    }

    public void setNumeroInscricao(String numeroInscricao) {
        this.numeroInscricao = numeroInscricao;
    }

    public String getNomeCandidato() {
        return nomeCandidato;
    }

    public void setNomeCandidato(String nomeCandidato) {
        this.nomeCandidato = nomeCandidato;
    }

    public void setConcursos(List<Concurso> concursos) {
        this.concursos = concursos;
    }

    /**
     * Lista de provas.
     * @return lista de provas
     */
    public List<Prova> getProvas() {
        if (concursoSelecionado != null && provas == null) {
            provas = digitacaoNotaManualFacade.consultarProvas(this.concursoSelecionado);
        }
        return provas;
    }

    public void setProvas(List<Prova> provas) {
        this.provas = provas;
    }

    public void setVersoes(List<VersaoResultadoProvaManual> versoes) {
        this.versoes = versoes;
    }

    public Uf getUfSelecionada() {
        return ufSelecionada;
    }

    public void setUfSelecionada(Uf ufSelecionada) {
        this.ufSelecionada = ufSelecionada;
    }

    public void setUfs(List<Uf> ufs) {
        this.ufs = ufs;
    }

    public Cidade getCidadeSelecionada() {
        return cidadeSelecionada;
    }

    public void setCidadeSelecionada(Cidade cidadeSelecionada) {
        this.cidadeSelecionada = cidadeSelecionada;
    }

    public void setCidades(List<Cidade> cidades) {
        this.cidades = cidades;
    }

    public List<Inscricao> getInscricoes() {
        return inscricoes;
    }

    public void setInscricoes(List<Inscricao> inscricoes) {
        this.inscricoes = inscricoes;
    }

    public Map<Inscricao, Boolean> getInscricoesExcluiveis() {
        return inscricoesExcluiveis;
    }

    public void setInscricoesExcluiveis(Map<Inscricao, Boolean> inscricoesExcluiveis) {
        this.inscricoesExcluiveis = inscricoesExcluiveis;
    }


    /**
     * Pesquisar candidatos habilitados para etapa/fase a qual pertence a prova,
     * observando todos os parâmetros informados.
     */
    public void pesquisar() {
        inscricoesExcluiveis = new HashMap<Inscricao, Boolean>();
        if (versaoSelecionada == VersaoResultadoProvaManual.PROVISORIA) {
            inscricoes = digitacaoNotaManualFacade.consultarInscricoes(provaSelecionada,
                orgaoSelecionado, cargoSelecionado, campoAtuacaoSelecionado,
                numeroInscricao, nomeCandidato, tipoConsultaNome);

            for (Inscricao inscricao : inscricoes) {
                inscricoesExcluiveis.put(inscricao, excluivel(inscricao));
            }
        } else {
            inscricoes = digitacaoNotaManualFacade.consultarInscricoesNotaDefinitiva(
                provaSelecionada, numeroInscricao, nomeCandidato, tipoConsultaNome);

            for (Inscricao inscricao : inscricoes) {
                inscricoesExcluiveis.put(inscricao, Boolean.FALSE);
            }
        }
    }

}
