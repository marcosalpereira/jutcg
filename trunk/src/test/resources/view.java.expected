package br.gov.esaf.sgc.view.correcao;

import java.util.HashMap;
import java.util.List;
import java.util.Map;
import javax.faces.component.UIComponent;
import javax.faces.component.UIInput;
import javax.faces.context.FacesContext;
import javax.faces.validator.ValidatorException;
import org.jboss.seam.ScopeType;
import org.jboss.seam.annotations.In;
import org.jboss.seam.annotations.Name;
import org.jboss.seam.annotations.Scope;
import br.gov.esaf.sgc.facade.DigitacaoNotaManualFacade;
import br.gov.esaf.sgc.model.CampoAtuacao;
import br.gov.esaf.sgc.model.Cargo;
import br.gov.esaf.sgc.model.Cidade;
import br.gov.esaf.sgc.model.Concurso;
import br.gov.esaf.sgc.model.Inscricao;
import br.gov.esaf.sgc.model.OrgaoContratante;
import br.gov.esaf.sgc.model.TipoConsultaNome;
import br.gov.esaf.sgc.model.Uf;
import br.gov.esaf.sgc.model.prova.Prova;
import br.gov.esaf.sgc.model.prova.ConteudoProva.TipoConteudoProva;
import br.gov.esaf.sgc.model.resultado.ResultadoProvaManual.VersaoResultadoProvaManual;
import br.gov.serpro.util.StringHelper;
import br.gov.serpro.view.util.MessageHelper;
import static org.easymock.EasyMock.*;
import org.junit.*;
import java.util.*;
import static org.junit.Assert.*;
import br.gov.esaf.sgc.view.JsfTestCase;

/**
 * Testes unitarios para a classe {@link DigitacaoNotaManualListarBean}.
 */
public class TestDigitacaoNotaManualListarBean extends JsfTestCase {

/**
 * Classe sendo testada {@link DigitacaoNotaManualListarBean}.
 */
private DigitacaoNotaManualListarBean digitacaoNotaManualListarBean;

/**
 * Cria o mock {@link DigitacaoNotaManualFacade} e seta na classe sendo testada.
 * @return o mock criado
 */
private DigitacaoNotaManualFacade criarMockDigitacaoNotaManualFacade() {
  DigitacaoNotaManualFacade mock = createStrictMock(DigitacaoNotaManualFacade.class);
  digitacaoNotaManualListarBean.setDigitacaoNotaManualFacade(mock);
  return mock;
}

/** {@inheritDoc} */
@Override
public void setUp() throws Exception {
  super.setUp();
  digitacaoNotaManualListarBean = new DigitacaoNotaManualListarBean();
}

/**
 * Teste para o metodo {@link DigitacaoNotaManualListarBean#pesquisar()}.
 * Descricao do Fluxo: 
 *     Entra: if (versaoSelecionada == VersaoResultadoProvaManual.PROVISORIA) {
 *     Entra: for (Inscricao inscricao : inscricoes) {.
 */
@Test
public void testPesquisarPath1() {
// variaveis usadas
final VersaoResultadoProvaManual versaoSelecionada = new VersaoResultadoProvaManual(1);
final TipoConsultaNome tipoConsultaNome = new TipoConsultaNome(1);
final Prova provaSelecionada = new Prova(1);
final OrgaoContratante orgaoSelecionado = new OrgaoContratante(1);
final String numeroInscricao = "1";
final String nomeCandidato = "1";
final List<Inscricao> inscricoes = Arrays.asList(new Inscricao(1));
final Cargo cargoSelecionado = new Cargo(1);
final CampoAtuacao campoAtuacaoSelecionado = new CampoAtuacao(1);

// Configurando mock
DigitacaoNotaManualFacade digitacaoNotaManualFacade = criarMockDigitacaoNotaManualFacade();
expect(digitacaoNotaManualFacade.consultarInscricoes(provaSelecionada,orgaoSelecionado,cargoSelecionado,campoAtuacaoSelecionado,numeroInscricao,nomeCandidato,tipoConsultaNome))
  .andReturn(inscricoes);
replay(digitacaoNotaManualFacade);

// Configurando estado interno da classe sob teste
digitacaoNotaManualListarBean.setVersaoSelecionada(versaoSelecionada);
digitacaoNotaManualListarBean.setTipoConsultaNome(tipoConsultaNome);
digitacaoNotaManualListarBean.setProvaSelecionada(provaSelecionada);
digitacaoNotaManualListarBean.setOrgaoSelecionado(orgaoSelecionado);
digitacaoNotaManualListarBean.setNumeroInscricao(numeroInscricao);
digitacaoNotaManualListarBean.setNomeCandidato(nomeCandidato);
digitacaoNotaManualListarBean.setInscricoes(inscricoes);
digitacaoNotaManualListarBean.setCargoSelecionado(cargoSelecionado);
digitacaoNotaManualListarBean.setCampoAtuacaoSelecionado(campoAtuacaoSelecionado);

// invocar metodo sendo testado
digitacaoNotaManualListarBean.pesquisar();

// checar estados dos mocks
verify(digitacaoNotaManualFacade);

// verificacoes do resultado do metodo sendo testado
assertEquals(expected, digitacaoNotaManualListarBean.getInscricoesExcluiveis());
assertEquals(expected, digitacaoNotaManualListarBean.getInscricoes());
}

/**
 * Teste para o metodo {@link DigitacaoNotaManualListarBean#pesquisar()}.
 * Descricao do Fluxo: 
 *     Entra: if (versaoSelecionada == VersaoResultadoProvaManual.PROVISORIA) {
 * NAO Entra: for (Inscricao inscricao : inscricoes) {.
 */
@Test
public void testPesquisarPath2() {
// variaveis usadas
final VersaoResultadoProvaManual versaoSelecionada = new VersaoResultadoProvaManual(1);
final TipoConsultaNome tipoConsultaNome = new TipoConsultaNome(1);
final Prova provaSelecionada = new Prova(1);
final OrgaoContratante orgaoSelecionado = new OrgaoContratante(1);
final String numeroInscricao = "1";
final String nomeCandidato = "1";
final List<Inscricao> inscricoes = Arrays.asList(new Inscricao(1));
final Cargo cargoSelecionado = new Cargo(1);
final CampoAtuacao campoAtuacaoSelecionado = new CampoAtuacao(1);

// Configurando mock
DigitacaoNotaManualFacade digitacaoNotaManualFacade = criarMockDigitacaoNotaManualFacade();
expect(digitacaoNotaManualFacade.consultarInscricoes(provaSelecionada,orgaoSelecionado,cargoSelecionado,campoAtuacaoSelecionado,numeroInscricao,nomeCandidato,tipoConsultaNome))
  .andReturn(inscricoes);
replay(digitacaoNotaManualFacade);

// Configurando estado interno da classe sob teste
digitacaoNotaManualListarBean.setVersaoSelecionada(versaoSelecionada);
digitacaoNotaManualListarBean.setTipoConsultaNome(tipoConsultaNome);
digitacaoNotaManualListarBean.setProvaSelecionada(provaSelecionada);
digitacaoNotaManualListarBean.setOrgaoSelecionado(orgaoSelecionado);
digitacaoNotaManualListarBean.setNumeroInscricao(numeroInscricao);
digitacaoNotaManualListarBean.setNomeCandidato(nomeCandidato);
digitacaoNotaManualListarBean.setInscricoes(inscricoes);
digitacaoNotaManualListarBean.setCargoSelecionado(cargoSelecionado);
digitacaoNotaManualListarBean.setCampoAtuacaoSelecionado(campoAtuacaoSelecionado);

// invocar metodo sendo testado
digitacaoNotaManualListarBean.pesquisar();

// checar estados dos mocks
verify(digitacaoNotaManualFacade);

// verificacoes do resultado do metodo sendo testado
assertEquals(expected, digitacaoNotaManualListarBean.getInscricoesExcluiveis());
assertEquals(expected, digitacaoNotaManualListarBean.getInscricoes());
}

/**
 * Teste para o metodo {@link DigitacaoNotaManualListarBean#pesquisar()}.
 * Descricao do Fluxo: 
 * NAO Entra: if (versaoSelecionada == VersaoResultadoProvaManual.PROVISORIA) {
 *     Entra: for (Inscricao inscricao : inscricoes) {.
 */
@Test
public void testPesquisarPath3() {
// variaveis usadas
final VersaoResultadoProvaManual versaoSelecionada = new VersaoResultadoProvaManual(1);
final TipoConsultaNome tipoConsultaNome = new TipoConsultaNome(1);
final Prova provaSelecionada = new Prova(1);
final OrgaoContratante orgaoSelecionado = new OrgaoContratante(1);
final String numeroInscricao = "1";
final String nomeCandidato = "1";
final List<Inscricao> inscricoes = Arrays.asList(new Inscricao(1));
final Cargo cargoSelecionado = new Cargo(1);
final CampoAtuacao campoAtuacaoSelecionado = new CampoAtuacao(1);

// Configurando mock
DigitacaoNotaManualFacade digitacaoNotaManualFacade = criarMockDigitacaoNotaManualFacade();
expect(digitacaoNotaManualFacade.consultarInscricoesNotaDefinitiva(provaSelecionada,numeroInscricao,nomeCandidato,tipoConsultaNome))
  .andReturn(inscricoes);
replay(digitacaoNotaManualFacade);

// Configurando estado interno da classe sob teste
digitacaoNotaManualListarBean.setVersaoSelecionada(versaoSelecionada);
digitacaoNotaManualListarBean.setTipoConsultaNome(tipoConsultaNome);
digitacaoNotaManualListarBean.setProvaSelecionada(provaSelecionada);
digitacaoNotaManualListarBean.setOrgaoSelecionado(orgaoSelecionado);
digitacaoNotaManualListarBean.setNumeroInscricao(numeroInscricao);
digitacaoNotaManualListarBean.setNomeCandidato(nomeCandidato);
digitacaoNotaManualListarBean.setInscricoes(inscricoes);
digitacaoNotaManualListarBean.setCargoSelecionado(cargoSelecionado);
digitacaoNotaManualListarBean.setCampoAtuacaoSelecionado(campoAtuacaoSelecionado);

// invocar metodo sendo testado
digitacaoNotaManualListarBean.pesquisar();

// checar estados dos mocks
verify(digitacaoNotaManualFacade);

// verificacoes do resultado do metodo sendo testado
assertEquals(expected, digitacaoNotaManualListarBean.getInscricoesExcluiveis());
assertEquals(expected, digitacaoNotaManualListarBean.getInscricoes());
}

/**
 * Teste para o metodo {@link DigitacaoNotaManualListarBean#pesquisar()}.
 * Descricao do Fluxo: 
 * NAO Entra: if (versaoSelecionada == VersaoResultadoProvaManual.PROVISORIA) {
 * NAO Entra: for (Inscricao inscricao : inscricoes) {.
 */
@Test
public void testPesquisarPath4() {
// variaveis usadas
final VersaoResultadoProvaManual versaoSelecionada = new VersaoResultadoProvaManual(1);
final TipoConsultaNome tipoConsultaNome = new TipoConsultaNome(1);
final Prova provaSelecionada = new Prova(1);
final OrgaoContratante orgaoSelecionado = new OrgaoContratante(1);
final String numeroInscricao = "1";
final String nomeCandidato = "1";
final List<Inscricao> inscricoes = Arrays.asList(new Inscricao(1));
final Cargo cargoSelecionado = new Cargo(1);
final CampoAtuacao campoAtuacaoSelecionado = new CampoAtuacao(1);

// Configurando mock
DigitacaoNotaManualFacade digitacaoNotaManualFacade = criarMockDigitacaoNotaManualFacade();
expect(digitacaoNotaManualFacade.consultarInscricoesNotaDefinitiva(provaSelecionada,numeroInscricao,nomeCandidato,tipoConsultaNome))
  .andReturn(inscricoes);
replay(digitacaoNotaManualFacade);

// Configurando estado interno da classe sob teste
digitacaoNotaManualListarBean.setVersaoSelecionada(versaoSelecionada);
digitacaoNotaManualListarBean.setTipoConsultaNome(tipoConsultaNome);
digitacaoNotaManualListarBean.setProvaSelecionada(provaSelecionada);
digitacaoNotaManualListarBean.setOrgaoSelecionado(orgaoSelecionado);
digitacaoNotaManualListarBean.setNumeroInscricao(numeroInscricao);
digitacaoNotaManualListarBean.setNomeCandidato(nomeCandidato);
digitacaoNotaManualListarBean.setInscricoes(inscricoes);
digitacaoNotaManualListarBean.setCargoSelecionado(cargoSelecionado);
digitacaoNotaManualListarBean.setCampoAtuacaoSelecionado(campoAtuacaoSelecionado);

// invocar metodo sendo testado
digitacaoNotaManualListarBean.pesquisar();

// checar estados dos mocks
verify(digitacaoNotaManualFacade);

// verificacoes do resultado do metodo sendo testado
assertEquals(expected, digitacaoNotaManualListarBean.getInscricoesExcluiveis());
assertEquals(expected, digitacaoNotaManualListarBean.getInscricoes());
}

/**
 * Teste para o metodo {@link DigitacaoNotaManualListarBean#excluivel(Inscricao)}.
 */
@Test
public void testExcluivel() {
// variaveis usadas
final Prova provaSelecionada = new Prova(1);
final Cidade cidadeSelecionada = new Cidade(1);
final boolean existeNotaProvisoriaStatusCadastradaFromMock = false;
final boolean existeNotaProvisoriaStatusCadastradaEsperado = false;
final Inscricao inscricao = new Inscricao(1);

// Configurando mock
DigitacaoNotaManualFacade digitacaoNotaManualFacade = criarMockDigitacaoNotaManualFacade();
expect(digitacaoNotaManualFacade.existeNotaProvisoriaStatusCadastrada(inscricao,provaSelecionada,cidadeSelecionada))
  .andReturn(existeNotaProvisoriaStatusCadastradaFromMock);
replay(digitacaoNotaManualFacade);

// Configurando estado interno da classe sob teste
digitacaoNotaManualListarBean.setProvaSelecionada(provaSelecionada);
digitacaoNotaManualListarBean.setCidadeSelecionada(cidadeSelecionada);

// invocar metodo sendo testado
final boolean excluivelReal = digitacaoNotaManualListarBean.excluivel(inscricao);

// checar estados dos mocks
verify(digitacaoNotaManualFacade);

// verificacoes do resultado do metodo sendo testado
assertEquals(existeNotaProvisoriaStatusCadastradaEsperado, excluivelReal);
}

/**
 * Teste para o metodo {@link DigitacaoNotaManualListarBean#excluirResultado(Inscricao)}.
 */
@Test
public void testExcluirResultado() {
// variaveis usadas
final Prova provaSelecionada = new Prova(1);
final Cidade cidadeSelecionada = new Cidade(1);
final Inscricao inscricao = new Inscricao(1);

// Configurando mock
DigitacaoNotaManualFacade digitacaoNotaManualFacade = criarMockDigitacaoNotaManualFacade();
digitacaoNotaManualFacade.excluirResultado(inscricao,provaSelecionada,cidadeSelecionada);
replay(digitacaoNotaManualFacade);

// Configurando estado interno da classe sob teste
digitacaoNotaManualListarBean.setProvaSelecionada(provaSelecionada);
digitacaoNotaManualListarBean.setCidadeSelecionada(cidadeSelecionada);

// invocar metodo sendo testado
digitacaoNotaManualListarBean.excluirResultado(inscricao);

// checar estados dos mocks
verify(digitacaoNotaManualFacade);
}
}
